src/IronPythonExtension/IronPythonExtension.cs

```diff
@@ -37,15 +37,15 @@ public void Dispose()
         /// <param name="sp"></param>
         public void Startup(StartupParams sp)
         {
-            // Searches for IronPython engine binary in Dynamo folder
-            var dynamoDir = Environment.CurrentDirectory;
+            // Searches for DSIronPython engine binary in same folder with extension itself
+            var targetDir = Path.GetDirectoryName(Assembly.GetAssembly(typeof(IronPythonExtension)).Location);
             var libraryLoader = sp.LibraryLoader;
             Assembly pythonEvaluatorLib = null;
             try
             {
-                pythonEvaluatorLib = Assembly.LoadFrom(Path.Combine(dynamoDir, PythonEvaluatorAssembly + ".dll"));
+                pythonEvaluatorLib = Assembly.LoadFrom(Path.Combine(targetDir, PythonEvaluatorAssembly + ".dll"));
             }
-            catch (Exception)
+            catch
             {
                 // Most likely the IronPython engine is excluded in this case
```

Comment:
@QilongTang would it be worth logging this? You can implement the ILogSource interface and log to the dynamo console quite easily from an extension if you wish.

Reply:
for example Docs browser does this:
https://github.com/DynamoDS/Dynamo/blob/5b086a92cab91dfead7ab6ec02862872fe8fcc3a/src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.cs#L43

Reply:
@mjkkirschner I have included the implementation but from my testing it does not work for extension?